app/Jobs/RetrieveIpAddressGeoLocationForSignature.php

```diff
@@ -58,17 +60,22 @@ public function handle(): void
             ]
         );
 
-        $response = $client->get(
-            $this->signature->ip_address,
-            [
-                'query' => [
-                    'access_key' => config('ipstack.api_key'),
-                    'output' => 'json',
-                ],
-            ]
-        );
+        try {
+            $response = $client->get(
+                $this->signature->ip_address,
+                [
+                    'query' => [
+                        'access_key' => config('ipstack.api_key'),
+                        'output' => 'json',
+                    ],
+                ]
+            );
 
-        $this->signature->ip_address_location_estimate = json_decode($response->getBody()->getContents());
-        $this->signature->save();
+            $this->signature->ip_address_location_estimate = json_decode($response->getBody()->getContents());
+            $this->signature->save();
+        } catch (ClientException $e) {
```

Comment:
I would rather just bubble up the exception, you get more info in horizon that way

Reply:
That doesn't help local dev though, because when the queue driver is `sync` the exception gets rendered to the user and then you have to restart the dues flow.

Reply:
Compromise - leave the handler, and I'll still pass the exception up via the failure handler.